### PR TITLE
Add mutation for $ -> \z

### PR DESCRIPTION
**Pull request by dgollahon**
Addresses #572

### CODE REVIEW

File: lib/mutant/mutator/node/generic.rb [hunk 1]
```diff
@@ -37,17 +37,18 @@ class Generic < self
         ]
 
         unsupported_regexp_nodes = AST::Types::REGEXP.to_a - %i[
-          regexp_root_expression
+          regexp_alternation_meta
           regexp_bol_anchor
-          regexp_word_type
-          regexp_nonword_type
           regexp_digit_type
+          regexp_eol_anchor
```
Comment (dgollahon) [inline lib/mutant/mutator/node/generic.rb]: This one was added, but I decided to sort the rest of the list because I was not sure where to insert it.

**Comment by dgollahon** [comment]
@mbj: Can you review?

File: meta/regexp/regexp_eol_anchor.rb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+Mutant::Meta::Example.add :regexp_eol_anchor do
+  source '/$/'
+
+  singleton_mutations
+  regexp_mutations
+
+  mutation '/\\z/'
```
Comment (backus) [inline meta/regexp/regexp_eol_anchor.rb]: Can this just be a single `\`?

Comment (dgollahon) [inline meta/regexp/regexp_eol_anchor.rb]: Yeah :+1:

**Comment by backus** [comment]
@dgollahon this is certainly not a requirement, but what I usually like to do when I'm adding new mutations which require an entirely new set of examples in the meta, is add an initial commit (before adding any source code) with the previous mutations for that source. I then follow that commit with my implementation. See the two commits on #633 as an example

**Comment by dkubb** [comment]
LGTM. Will merge on green CI.

**Pull request merged**